apiary.apib

```diff
@@ -1009,20 +1009,1228 @@ Outputs the list of peers that were removed.
+
+        curl -i "http://localhost:5001/api/v0/object/new?arg=kitten"
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/object/new?arg="
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/object/new?arg="
+        ```
+
++ Response 500
+
+    + Headers
+
+        ```
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 04 Feb 2016 14:50:39 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: "template '' not found"
+        - Code: 0
+
+    + Body
+
+        ```
+        {
+          "Message": "template '' not found",
+          "Code": 0
+        }
+        ```
+
++ Request With Argument
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/object/new?arg=unixfs-dir"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/object/new?arg=unixfs-dir"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 04 Feb 2016 14:52:31 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (object)
+        - Hash (Multihash)
+        - Links () - // TODO What should this be?
+
+    + Body
+
+        ```
+        {
+          "Hash": "QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn",
+          "Links": null
+        }
+        ```
+
++ Request With Argument And XML Encoding
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/object/new?arg=unixfs-dir&encoding=xml"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/object/new?arg=unixfs-dir&encoding=xml"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Content-Type: application/xml
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 04 Feb 2016 14:52:31 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        <Object><Hash>QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn</Hash></Object>
+        ```
+
+## patch [GET /object/patch{?arg1,arg2}{&name,ref}]
+Create a new merkledag object based on an existing one
+
+'patch' adds and removes links from objects, creating a new object as a result. This is the merkle-dag version of modifying an object. It can also set the data inside a node with `set-data` and append to that data as well with `append-data`.
+
++ Parameters
+    + arg1 (string, required) - the hash of the node to modify
+    + arg2 (enum[string], required) - the operation to perform
+        + Members:
+            + `add-link` - with `<name> <ref>`, adds a link to a node
+            + `rm-link` - with `<name>`, removes a link from a node
+            + `set-data` -  sets a node's data
+            + `append-data` - appends to a node's data
```

Comment:
this is 0.3.11 only, in 0.4.0 these args became the name of the commands //cc @dignifiedquire